RELEASES.md

```diff
@@ -0,0 +1,55 @@
+# Releases
+
+This document describes how Strimzi handles versioning, releases new versions and communicates about them.
+
+Strimzi uses [semantic versioning](https://semver.org/) as a versioning schema for the releases of all the components.
+
+## Release Cycle
+
+Strimzi is using a release schedule of 1 or 2 months for the "core" component, the cluster operator.
+The [GitHub milestones](https://github.com/strimzi/strimzi-kafka-operator/milestones) are used to track the features and bug fixes included in each release.
+There is also a [roadmap](https://github.com/orgs/strimzi/projects/1) to describe the features that will be available in the future releases.
+
+The other components, under the Strimzi umbrella, are released on a case-by-case basis depending on the new features added, bugs and vulnerabilities fixed.
+
+## Release Process
+
+Most of the Strimzi components have a corresponding documentation about how to make a new release.
+For example, the release process for the cluster operator is described [here](https://github.com/strimzi/strimzi-kafka-operator/blob/main/development-docs/RELEASE.md).
+
+Before an official release, at least one Release Candidate (RC) is built for testing.
+More RCs can follow the first one if new bugs are found along the way.
+The Release Candidates are announced on the Slack channel and Strimzi user CNCF mailing list, so that the community can test before the final release.
+The maintainers provides 2-3 days to the community for testing and, if no objections are raised, the final release is announced.
```

Comment:
Comparing this to the [Kafka release process](https://cwiki.apache.org/confluence/display/KAFKA/Release+Process) (see also [the Kafka bylaws](https://cwiki.apache.org/confluence/display/KAFKA/Bylaws)), it's not so clear _who_ gets to decide about the validity of any objections. In fact, taken literally, any random person/bot on the internet could "object" and prevent a release. I'm not sure we want/need to document every last detail, but it would be good to be _bit_ clearer. For example, making the actual final release dependent on a maintainer vote makes it clearer who decides about any objections, and I don't think it would slow down releases very much, because we already seek a consensus on releases, just not publicly enough.

Reply:
So are you suggesting to add a sentence like ...

"If any objections, maybe related to a bug, there will be a vote from maintainers to decide to stop or move forward with the release". or "If any objections, maybe related to a bug, the maintainers have to reach a consensus on stopping or moving forward with the release".

Or do you want something more specific about how it works, via email, slack channel ... ?

Reply:
Vote suggests a formal process which you would need to define. We should either really setup a proper release voting process. So I guess you can say something like _any objections will be evaluated and decided on_?

Not sure anyone would raise it in the incubation - but of course Jakub Scholz announcing RC and without any sort of feedback announcing GA is not exactly a process. A process following Apache style voting as for example used in Kafka would be obviously more advanced. But the question is whether we have enough resources to change it.

Reply:
> What resources does it need other than maintainers paying attention?

The maintainers and their time are the resources here.